lib/matplotlib/image.py

```diff
@@ -417,18 +430,10 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
                 if a_min != a_max:
                     A_scaled /= ((a_max - a_min) / 0.8)
                 A_scaled += 0.1
-                A_resampled = np.zeros((out_height, out_width),
-                                       dtype=A_scaled.dtype)
                 # resample the input data to the correct resolution and shape
-                _image.resample(A_scaled, A_resampled,
-                                t,
-                                _interpd_[self.get_interpolation()],
-                                self.get_resample(), 1.0,
-                                self.get_filternorm(),
-                                self.get_filterrad())
-
-                # we are done with A_scaled now, remove from namespace
-                # to be sure!
+                A_resampled = _resample(self, A_scaled, out_shape, t)
```

Comment:
Since you are always passing `self` as first parameter, wouldn't it make sense to make `_resample` an instance method?

Reply:
I think of it more as a wrapper to _image.resample than as a method of ImageBase, so I'd rather leave it here, but I can change that if you insist.

Reply:
That's ok then.